build/template_kql/common/extend_resource.kql

```diff
@@ -307,7 +329,21 @@ resources
     (type == 'microsoft.documentdb/databaseaccounts' and properties.enableAutomaticFailover == "true"), "Configured",
     (type == 'microsoft.documentdb/databaseaccounts'), "Not Applicable",
     type == 'microsoft.dbformysql/servers', "Not Applicable",
-    type == 'microsoft.dbformysql/flexibleservers', case(properties.haEnabled == "Enabled", "Configured", "Not Applicable"),
+     type == 'microsoft.dbformysql/flexibleservers', case(
+        properties.highAvailability.mode == "Disabled" and properties.replicationRole != "Replica" , "Not Configured",
+        properties.replicationRole == "Replica", "Not Applicable",
+        properties.highAvailability.mode == "ZoneRedundant", "Zone Redundant",
+        properties.highAvailability.mode == "SameZone", "Same Zone",
+        properties.highAvailability.mode
+    ),
+    type == 'microsoft.dbforpostgresql/servers', "Not Applicable",
+    type == 'microsoft.dbforpostgresql/flexibleservers', case(
+        properties.highAvailability.mode == "Disabled" and properties.replicationRole != "Replica" , "Not Configured",
+        properties.replicationRole == "Replica", "Not Applicable",
+        properties.highAvailability.mode == "ZoneRedundant", "Zone Redundant",
+        properties.highAvailability.mode == "SameZone", "Same Zone",
+        properties.highAvailability.mode
+    ),
```

Comment:
You need to update this part as follows because PostgreSQL replications role is different from MySQL:
```
    type == 'microsoft.dbforpostgresql/flexibleservers', case(
        properties.highAvailability.mode == "Disabled" and properties.replicationRole != "GeoAsyncReplica" , "Not Configured",
        properties.replicationRole == "GeoAsyncReplica", "Not Applicable",
        properties.highAvailability.mode == "ZoneRedundant", "Zone Redundant",
        properties.highAvailability.mode == "SameZone", "Same Zone",
        properties.highAvailability.mode
    ),
```

Reply:
Thank you this is updated.